server/src/Api/Extensions/FormFileExtentions.cs

```diff
@@ -0,0 +1,24 @@
+using Microsoft.AspNetCore.Http;
+using StudyZen.Api.Exceptions;
+using StudyZen.Application.Services; 
+
+namespace StudyZen.Api.Extensions
+{
+    public static class FormFileExtensions
+    {
+        public static void ImportFlashcardsFromCsvStream(this IFormFile file, FlashcardFileImporter importer, int flashcardSetId)
+        {
+            try
+            {
+                using (var stream = file.OpenReadStream())
+                {
+                    importer.ImportFlashcardsFromCsvStream(stream, flashcardSetId);
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new ImportFailedException("Error importing flashcards.", ex);
```

Comment:
Why catch and throw a new exception in this case? You are catching an Exception object, which means that all exceptions will gather here. For example, here could be a server error or user error and it will still be HTTP 400.